### DISCUSSION TITLE
v2.1.4

### DISCUSSION THREAD

**Discussion by gillett-hernandez** [gillett-hernandez/Foreman2#9] [Announcements]
* Fix divide by zero error where assembler speed is 0. this is not a permanent fix but just one to force foreman to allow you to change away from the offending assembler.
* Fix localization issue so now languages other than english can be imported. also fixes the bugged out characters that appeared in place of uncommon symbols (° for example). note this does not add localization to foreman2, in terms of the menu language of that. it merely leverages factorio's localization for recipe names, item names, etc.

<hr /><em>This discussion was created from the release <a href='https://github.com/gillett-hernandez/Foreman2/releases/tag/'>v2.1.4</a>.</em>

**Comment by PG-Fuury** [comment]
Assembler speed being 0 can be avoided by clamping the speed at 20%, as the game does (at least that happens with Space Exploration, not sure if other mods with high tier productivity modules interact differently in that way, but it seems like an engine-level thing)

**Reply by gillett-hernandez** [reply]
I already clamp speed at a minimum of 0.2 in when calculating module effects, this change was more about handling mods that for whatever reason have assemblers with a speed set to 0. As an example I'm pretty sure the condenser turbines in SE have a speed set to 0, or some of the nuclear reactors.
